aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -156,12 +168,22 @@ Tensor empty_like(const Tensor& self) {
   return native::empty_like(self, self.options());
 }
 
-Tensor empty_like(const Tensor& self, const TensorOptions& options) {
+Tensor empty_like(
+    const Tensor& self,
+    const TensorOptions& options,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
+
+  TORCH_CHECK(
+      !(options.layout() == kSparse &&
+          optional_memory_format.has_value()),
+      " memory format options is incompatible with sparse tensors");
   if (options.layout() == kSparse && self.is_sparse()) {
-    auto res = at::empty({0}, options); // to be resized
-    res.sparse_resize_and_clear_(self.sizes(), self.sparse_dim(), self.dense_dim());
-    return res;
+    auto result = at::empty({0}, options); // to be resized
+    result.sparse_resize_and_clear_(
+        self.sizes(), self.sparse_dim(), self.dense_dim());
+    return result;
   }
+
   if (self.is_quantized()) {
```

Comment:
Memory format ought to apply to quantized tensors, no?

Reply:
I'm afraid we can't empty_like quantized tensors ATM.
Master:
```
q = torch._empty_affine_quantized([10,3,16,16], scale=0.5, zero_point=10, dtype=torch.quint8)   
z = torch.empty_like(q)     
```

`Must not create a new variable from a variable, use its .tensor_data()`

Reply:
@jerryzh168 This sounds wrong. Do you know why this is happening?

Reply:
not sure, I added this branch because empty_like is called in `contiguous`: https://github.com/pytorch/pytorch/pull/20869

Reply:
So I think C++ should work, and there might be something wrong with variables

Reply:
This is using the wrong options, see https://github.com/pytorch/pytorch/pull/21978.

Reply:
Will merge in as soon as landed.